apkid/rules/elf/obfuscators.yara

```diff
@@ -130,7 +132,87 @@ rule ollvm_v6_0 : obfuscator
     $based_on      = "(based on Obfuscator-LLVM 6.0."
 
   condition:
-    all of them and not ollvm_v6_0_strenc
+    is_elf and
+    all of them and
+    not ollvm_v6_0_strenc
+}
+
+rule ollvm_v9_strenc : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 9.x (string encryption)"
+    sample      = "2314ec0053d829d424a82f702188fcb525cefce4feeef096f0855339b897a5d1"
+    author      = "Eduardo Novella"
+
+  strings:
+    $clang_version = /clang version \d\.\d\.\d /
+    $strenc        = /\.datadiv_decode[\d]{18,20}/
+    $ollvm         = "(based on Obfuscator-LLVM 9."
+
+  condition:
+    is_elf and all of them
+}
+
+rule ollvm_tll : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM TLL (string encryption)"
+    url         = "https://github.com/yazhiwang/ollvm-tll"
+    sample      = "1f010330e9ac90f00d11aa37fdca25c437ad6f4b1302f6d7aa48b91ef22cc107"
+    author      = "Eduardo Novella"
+
+  strings:
+    /**
+      .datadiv_decode7760209850571766755
+      Android clang version 5.0.300080  (based on LLVM 5.0.300080)
+      clang version 6.0.0 (tags/RELEASE_600/final) (https://github.com/yazhiwang/ollvm-tll.git a38559e4c13359073102793c0a734bb1add3d5ff)
+    */
+    $clang_version = /clang version \d\.\d\.\d \(tags\/RELEASE\_\d+\/final\)/
+    $strenc        = /\.datadiv_decode[\d]{18,20}/
+    $url           = "https://github.com/yazhiwang/ollvm-tll"
+
+  condition:
+    is_elf and all of them
```

Comment:
Same as above.

So, is the only thing we're differentiating upon for these the clang version, then specifically for this one the url?

Reply:
Why was this marked as resolved? I have the same question. Are these really different variants or different versions?

Having different variants makes sense when it's different versions of the tool. I.e. one variant could mean they're using a really old tool, or it's an old sample, while a different variant would imply they're using more up to date tools. This information is hard to get, but ideally it should be used. It's not obvious why these different rules exist when the strings we're looking for seem to imply that they're the same version.

If all of these rules are to cover a single version or very very similar versions of a tool, then this could be 1-2 rules.